message/views.py

```diff
@@ -1,3 +1,33 @@
-# from django.shortcuts import render
+from django.shortcuts import render, redirect
+from game_event.models import GameEvent
+from player.models import Player
+from .models import Message
 
-# Create your views here.
+
+def save_text_message(request, id):
+    try:
+        event = GameEvent.objects.get(pk=id)
+    except GameEvent.DoesNotExist:
+        return render(request, 'game_event/game-event.html', {})
+    try:
+        player = Player.objects.get(user=request.user)
+    except Player.DoesNotExist:
```

Comment:
i would still be able to alter the id value at the url of the browser if i am not wrong.
so the login decorator is a good addition but not an alternative

Reply:
@Danielsio If I understand correctly, the `id `referred to here is the event ID, not the user ID (which is obtained from the URL). However, you raise a valid concern that a user could potentially modify the event ID in the URL and send a message to an event they are not actually a part of. Perhaps it would be wise to implement a verification step before sending the message to ensure that the player is indeed a member of the event

Reply:
> @Danielsio If I understand correctly, the `id `referred to here is the event ID, not the user ID (which is obtained from the URL). However, you raise a valid concern that a user could potentially modify the event ID in the URL and send a message to an event they are not actually a part of. Perhaps it would be wise to implement a verification step before sending the message to ensure that the player is indeed a member of the event

Hi, just see your comment after I push the changes. So do you think I need to do some verification about it?

Reply:
I will add this verification, you're right.

Reply:
By using `@login_required`, we ensure that only logged-in players can access the app. The login and register parts already have tests, so we can rely on them.

But we need to verify the player's participation in the event.

Reply:
> By using `@login_required`, we ensure that only logged-in players can access the app. The login and register parts already have tests, so we can rely on them.
> 
> But we need to verify the player's participation in the event.

I added verification and modify the test with invalid player